pandas/core/generic.py

```diff
@@ -2079,6 +2079,8 @@ def __array__(self, dtype: npt.DTypeLike | None = None) -> np.ndarray:
             ):
                 arr = arr.view()
                 arr.flags.writeable = False
+            else:
+                arr.flags.writeable = True
```

Comment:
Why was `arr` not writeable to start with? Should that be fixed at the "source"? (i.e. Manager.as_array I suppose)

Because if for some reason you would have a non-writeable numpy array stored in the DataFrame, we would now override that flag in certain cases.